sunpy/io/tests/test_asdf.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+
+import asdf
+
+from sunpy.data.test import get_test_filepath
+from sunpy.io._asdf import get_header, get_keys_name, read, write
+from sunpy.io._header import FileHeader
+
+map_for_asdf = get_test_filepath("aiamap_genericmap_1.0.0.asdf")
+
+def test_read():
+    cont = read(map_for_asdf)
+    assert isinstance(cont,list)
+    assert isinstance(cont[0][0],np.ndarray)
+    assert isinstance(cont[0][1],FileHeader)
+    assert cont[0][0].shape ==  (2,2)
+
+def test_write(tmpdir):
+    data, header = read(map_for_asdf)[0]
+    outfile = tmpdir / "test.asdf"
+    write(str(outfile), data, header)
+    assert outfile.exists()
```

Comment:
Can we check the written file is the same as the starting file?

Reply:
sure i have added that test (test_write_and_verify)